packages/svelte-parse/src/main.ts

```diff
@@ -0,0 +1,1219 @@
+import {
+	Point,
+	Node,
+	BaseSvelteTag,
+	SvelteTag,
+	Property,
+	SvelteElement,
+	Directive,
+	Text,
+	Literal,
+	Root,
+	SvelteExpression,
+	VoidBlock,
+	BranchingBlock,
+	Branch,
+	Comment,
+	Parent,
+	SvelteComponent,
+} from 'svast';
+
+import {
+	ParseNodeOptions,
+	ParseOptions,
+	Result,
+	State,
+	SLASH,
+	CLOSE_ANGLE_BRACKET,
+	COLON,
+	EQUALS,
+	QUOTE,
+	APOSTROPHE,
+	TAB,
+	CLOSE_BRACE,
+	BACKTICK,
+	BACKSLASH,
+	AT,
+	OCTOTHERP,
+	RE_BLOCK_BRANCH,
+	RE_SCRIPT_STYLE,
+	RE_COMMENT_START,
+	RE_COMMENT_END,
+	RE_END_TAG_START,
+} from './types_and_things';
+
+import {
+	LINEFEED,
+	OPEN_ANGLE_BRACKET,
+	OPEN_BRACE,
+	UPPERCASE_A,
+	UPPERCASE_Z,
+	LOWERCASE_A,
+	LOWERCASE_Z,
+	SPACE,
+	PIPE,
+} from './types_and_things';
+
+import { void_els } from './void_els';
+
+function is_in_range(number: number, from: number, to: number): boolean {
+	return number >= from && number <= to;
+}
+
+function is_upper_alpha(code: number) {
+	return is_in_range(code, UPPERCASE_A, UPPERCASE_Z);
+}
+
+function is_lower_alpha(code: number) {
+	return is_in_range(code, LOWERCASE_A, LOWERCASE_Z);
+}
+
+function is_void_element(tag_name: string): boolean {
+	return void_els.includes(tag_name);
+}
+
+export function parseNode(opts: ParseNodeOptions): Result | undefined {
+	let index = 0;
+	let quote_type = '';
+	let expr_quote_type = '';
+	let closing_tag_name = '';
+	let brace_count = 0;
+
+	const node_stack: Node[] = [];
+	const state: State[] = [];
+
+	const {
+		value,
+		currentPosition = {
+			line: 1,
+			column: 1,
+			offset: 0,
+		},
+		block = true,
+		childParser,
+		generatePositions = true,
+	} = opts;
+
+	let position = Object.assign(currentPosition, { index });
+	let char = value.charCodeAt(index);
+
+	function chomp() {
+		// newline means new line
+		if (char === LINEFEED) {
+			position.line++;
+			position.column = 1;
+		} else {
+			// otherwise shift along the column pointer
+			position.column++;
+		}
+
+		// refers to the current parse
+		index++;
+
+		//@ts-ignore
+		position.offset++;
+		// stay in sync
+		position.index = index;
+		char = value.charCodeAt(index);
+	}
+
+	function place() {
+		const _p = Object.assign({}, position);
+		delete _p.index;
+		return _p;
+	}
+
+	let current_state: State | undefined;
+
+	function pop_state() {
+		state.pop();
+		current_state = state[state.length - 1];
+	}
+
+	function set_state(name: State, toPop?: boolean) {
+		if (toPop) state.pop();
+		state.push((current_state = name));
+	}
+
+	let current_node: Node | undefined;
+
+	function push_node(node: Node) {
+		node_stack.push((current_node = node));
+	}
+
+	function pop_node() {
+		node_stack.pop();
+		current_node = node_stack[node_stack.length - 1];
+	}
+
+	for (;;) {
+		// console.log(value[index], node_stack, state);
+		if (!value[index]) {
```

Comment:
```suggestion
		if (value[index] == null) {
```

Reply:
but better to check `=== void 0` if you know that to be the only case (think so)